.reek

```diff
@@ -27,7 +27,7 @@ TooManyMethods:
     - Devise::TwoFactorAuthenticationController
     - Users::ConfirmationsController
     - ServiceProvider
-    - Idv::SessionsController
+    - Idv::Session
```

Comment:
I'm not sure moving all the methods from the controller to a class addresses the issue. The goal here is to have many small classes that each have a single responsibility. One way to determine if a class is doing too much is to describe it in a single sentence. If the sentence contains the words `and` or `or`, it's a sign that it might be doing too much.

Here's what Sandi Metz has to say about SRP and why it matters:

> Applications that are easy to change consist of classes that are easy to reuse. Reusable classes are pluggable units of well-defined behavior that have few entanglements. An application that is easy to change is like a box of building blocks; you can select just the pieces you need and assemble them in unanticipated ways. A class that has more than one responsibility is difficult to reuse. The various responsibilities are likely thoroughly entangled within the class. If you want to reuse some (but not all) of its behavior, it is impossible to get at only the parts you need. You are faced with two options and neither is particularly appealing. If the responsibilities are so coupled that you cannot use just the behavior you need, you could duplicate the code of interest. This is a terrible idea. Duplicated code leads to additional maintenance and increases bugs. If the class is structured such that you can access only the behavior you need, you could reuse the entire class. This just substitutes one problem for another. Because the class you're reusing is confused about what it does and contains several tangled up responsibilities, it has many reasons to change. It may change for a reason that is unrelated to your use of it, and each time it changes there's a possibility of breaking every class that depends on it. You increase your application's chance of breaking unexpectedly if you depend on classes that do too much.